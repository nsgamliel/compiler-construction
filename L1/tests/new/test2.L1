(:tensors
  (:tensors 0 0
  rdi <- 3
  rsi <- 7
  call allocate 2
  rdi <- rax
  call print 1
  rdi <- 3 //this is a comment
  rsi <- 5
  call allocate 2
  rdi <- 7
  rsi <- rax
  call allocate 2
  rdi <- 9
  rsi <- rax
  call allocate 2
  rdi <- rax
  call print 1
  r10 <- 123
  r11 <- 5
  rdi <- 5
  :start
  r10 >>= 3
  rdi += 3
  cjump r11 <= r10 :start
  call print 1
  rdi <- 15
  rdi *= 5
  rdi &= 31
  cjump rdi = 1023 :start
  call print 1
  return
  )
)