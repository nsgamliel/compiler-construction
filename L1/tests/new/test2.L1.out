{s:1, 3}
{s:4, {s:3, {s:1, 2}, {s:1, 2}, {s:1, 2}}, {s:3, {s:1, 2}, {s:1, 2}, {s:1, 2}}, {s:3, {s:1, 2}, {s:1, 2}, {s:1, 2}}, {s:3, {s:1, 2}, {s:1, 2}, {s:1, 2}}}
5
5
